src/test/java/io/codeka/gaia/config/security/SecurityConfigIT.java

```diff
@@ -1,31 +1,32 @@
-package io.codeka.gaia.config;
+package io.codeka.gaia.config.security;
 
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.stereotype.Controller;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.web.servlet.MockMvc;
 
 import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders.formLogin;
 import static org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers.authenticated;
 
-@SpringBootTest(classes = {EmptyController.class, SecurityConfig.class})
+@SpringBootTest(classes = SecurityConfig.class)
+@DirtiesContext
 @AutoConfigureMockMvc
 @TestPropertySource(properties = "gaia.admin-password=admin456")
-class SecurityConfigTest {
+class SecurityConfigIT {
 
     @Autowired
     private MockMvc mockMvc;
 
+    @MockBean
+    SuccessHandler successHandler;
+
     @Test
     void adminUserPassword_shouldBeConfigurable() throws Exception {
         mockMvc.perform(formLogin().user("admin").password("admin456"))
                 .andExpect(authenticated().withUsername("admin"));
     }
```

Comment:
You can add a small assertion for the successHandler?
It seems weird to add it to the test without adding an assertion :stuck_out_tongue: 
Maybe something simple like :
```java
verify(successHandler).onAuthenticationSuccess(any(), any(), any());
```